### PR TITLE
requirePaddingNewLinesAfterBlocks: add array and object exceptions

### PR DESCRIPTION
**Pull request by markelog**
Ref gh-1239

### CODE REVIEW

**Comment by mikesherov** [comment]
:+1:

**Comment by TheSavior** [comment]
:+1: Thanks

**Comment by markelog** [comment]
Added object exception

File: lib/rules/require-padding-newlines-after-blocks.js [hunk 1]
```diff
@@ -67,6 +49,27 @@ var excludes = {
...
 };
 
+var exceptions = {
+    'CallExpression': true,
+    'ArrayExpression': true,
+    'Property': true
+};
+
+function isException(parent) {
+    var parParent = parent.parentNode;
```
Comment (zxqfox) [inline lib/rules/require-padding-newlines-after-blocks.js]: grandpa?

**Comment by zxqfox** [comment]
LGTM

File: lib/rules/require-padding-newlines-after-blocks.js (hunk 1 above)
Comment (markelog) [inline lib/rules/require-padding-newlines-after-blocks.js]: okay :-)

File: lib/rules/require-padding-newlines-after-blocks.js [hunk 2]
```diff
@@ -67,6 +49,27 @@ var excludes = {
...
 };
 
+var exceptions = {
+    'CallExpression': true,
+    'ArrayExpression': true,
+    'Property': true
+};
+
+function isException(parent) {
+    var grandpa = parent.parentNode;
```
Comment (zxqfox) [inline lib/rules/require-padding-newlines-after-blocks.js]: :smile:

**Comment by TheSavior** [comment]
I **disagree** with the change to this rule that doesn't enforce the extra line in objects. That was one of the key things I **needed** when I wrote this rule. Airbnb's style guide agreed and approved that. Why was it removed here?

**Comment by zxqfox** [comment]
@TheSavior Thanks for coming. Perhaps we need not to restore functionality you needed here in separate rule to left rules simple and focused.

And also update presets to be consistent with new rules.

Thoughts?

**Comment by markelog** [comment]
Hm, after some thinking, i would prefer if we would merge https://github.com/jscs-dev/node-jscs/pull/1239 and this one together, while adding third option for ignoring object expressions.

How is that sound?

**Comment by TheSavior** [comment]
I'm fine with that. When you say merge the two together, what do you mean? Does this PR provide any other behavior? Or should we just add the third option to #1239 and merge that one?

**Comment by markelog** [comment]
I'm thinking we could move exceptions check to the `isException` helper

**Pull request closed without merging**